--- gtk+-1.2.10/gtk/gtkstyle.c.orig	2003-01-22 15:41:57.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkstyle.c	2003-01-22 16:27:31.000000000 +0100
@@ -326,11 +326,11 @@
 static GdkColor gtk_default_selected_fg =    { 0, 0xffff, 0xffff, 0xffff };
 static GdkColor gtk_default_insensitive_fg = { 0, 0x7530, 0x7530, 0x7530 };
 
-static GdkColor gtk_default_normal_bg =      { 0, 0xd6d6, 0xd6d6, 0xd6d6 };
-static GdkColor gtk_default_active_bg =      { 0, 0xc350, 0xc350, 0xc350 };
-static GdkColor gtk_default_prelight_bg =    { 0, 0xea60, 0xea60, 0xea60 };
-static GdkColor gtk_default_selected_bg =    { 0,      0,      0, 0x9c40 };
-static GdkColor gtk_default_insensitive_bg = { 0, 0xd6d6, 0xd6d6, 0xd6d6 };
+static GdkColor gtk_default_normal_bg =      { 0, 0xdcdc, 0xdada, 0xd5d5 };
+static GdkColor gtk_default_active_bg =      { 0, 0xbaba, 0xb5b5, 0xabab };
+static GdkColor gtk_default_prelight_bg =    { 0, 0xeeee, 0xebeb, 0xe7e7 };
+static GdkColor gtk_default_selected_bg =    { 0, 0x4b4b, 0x6969, 0x8383 };
+static GdkColor gtk_default_insensitive_bg = { 0, 0xdcdc, 0xdada, 0xd5d5 };
 
 GtkStyle*
 gtk_style_copy (GtkStyle *style)
